docs/sphinx/examples/cpp/algorithms/bernstein_vazirani.cpp

```diff
@@ -1,6 +1,6 @@
 // Compile and run with:
 // ```
-// nvq++ bernstein_vazirani.cpp -o bv.x --target nvidia-mgpu
+// nvq++ bernstein_vazirani.cpp -o bv.x --target nvidia --nvidia-option mgpu
```

Comment:
I can't quite put my finger on it, but `--nvidia-option` doesn't quite sound right to me. The whole command line argument is a command line "option", so it seems redundant to put it in the name.

Anyway, I don't have a better suggestion, and I realize we're trying to wrap up this release, so I realize we may not do anything about this, but I figured I'd make a note just in case it spurred any better naming ideas from anybody else.

Reply:
I think LLVM / MLIR passes follow the `--<name>-option` style. So, it will change per target. How about we call it generically `--target-option`? 
So this would be - ```nvq++ bernstein_vazirani.cpp -o bv.x --target nvidia --target-option mgpu```.

Reply:
> How about we call it generically `--target-option`?

Lol, I think I was coming to that idea later in the review :)

> I think LLVM / MLIR passes follow the --<name>-option style. 

I don't think I've seen that before. I've seen (and used) this style before, though: `--canonicalize='enable-patterns=AnyPattern region-simplify=aggressive'`. I'm always curious about the various ways to set those CLI options for LLVM/MLIR. Do you know of a good resource to bookmark on those? And are those `--<name>-option` or `--<name>-<option>`?

Reply:
I'd vote for `--target-option`. 

We have the convention that target-specific options are prefixed with the target name for filtering purposes (not sending to the compiler).
`--target-option` seems synonymous with `--<target name>-option`, hence would be reasonable to be added to the CLI options filter logic.

> I'd like to get rid of the -- prefixes in our driver to be more like other C++ compilers.
That's a good suggestion. We might circle back to it after wrapping up this release since this would be a breaking change regarding existing options.